### DISCUSSION TITLE
Expect arm64 vs x86_64 issue with trap WINCH

### DISCUSSION THREAD

**Discussion by gcstang** [Homebrew/discussions#3233] [Everyday usage]
When using expect arm64 the trap/WINCH doesn't capture rows and columns but the x86_64 version does on the m1, is this a compile or install issue?

Example in an expect script
`trap {`
` #fetch rows and cols from controlling terminal`
` set rows [stty rows]`
` set cols [stty columns]`
` send "echo size changed to $rows $cols\r"`
` stty rows $rows columns $cols < $spawn_out(slave,name)`
`} WINCH`

What I ended up doing to get this to work on an m1
Install Homebrew for x86_64
`arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"`
Install expect for x86_64
`arch -x86_64 brew install expect`

If I use the expect in /usr/local/bin/expect now it works properly.

**Comment by SMillerDev** [comment]
Did you use the homebrew expect on arm64? MacOS ships one too

**Comment by gcstang** [comment]
I used the home brew one only, didn't know Mac had one

**Comment by carlocab** [comment]
Sounds like https://github.com/Homebrew/homebrew-core/issues/77875.

**Comment by gcstang** [comment]
definitely does, thank you
